darwin/client.py

```diff
@@ -23,16 +23,16 @@
 
 class Client:
     def __init__(self, config: Config, default_team: Optional[str] = None):
-        self.config = config
-        self.url = config.get("global/api_endpoint")
-        self.base_url = config.get("global/base_url")
-        self.default_team = default_team or config.get("global/default_team")
-        self.features: dict = {}
+        self.config: Config = config
+        self.url: str = config.get("global/api_endpoint")
+        self.base_url: str = config.get("global/base_url")
+        self.default_team: str = default_team or config.get("global/default_team")
+        self.features: Dict[str, Any] = {}
```

Comment:
I wonder if we can do better than `Any`.

Can we define something like this in our `types.py`
```python
from dataclasses import dataclass

@dataclass
class FeaturePayload:
    """Structured payload of a Feature record on V7 Darwin"""
    name: str
    enabled: boolean
```

This way, we can define `self.features: Dict[str, FeaturePayload]`.

This method is very similar to what we do on the FE side (TypeScript there).

Note: the reason why I'm using a `dataclass` here is that we can't use `TypedDict`, which I believed was introduced in Python 3.8 if I'm not mistaken.

Note: we probably need to edit `load_feature_flags`. We currently do:
```python
self.features[team_slug] = self.get(f"/teams/{team_slug}/features")
```
which is terrible for many reasons.